lib/src/merged_tree.rs

```diff
@@ -705,23 +708,39 @@ impl Iterator for TreeDiffIterator<'_> {
                 },
                 TreeDiffItem::File(..) => {
                     if let TreeDiffItem::File(path, before, after) = self.stack.pop().unwrap() {
-                        return Some((path, Ok((before, after))));
+                        return Some(TreeDiffEntry {
+                            target: path,
+                            value: Ok((before, after)),
+                        });
                     } else {
                         unreachable!();
                     }
                 }
             };
 
+            let path = entry.target;
+            let (before, after) = entry.value.unwrap();
```

Comment:
Perhaps, we wouldn't have to change the type of `TreeDiffDirItem`. We don't need source/dest directory paths. Then, this `.unwrap()` can be removed.

Reply:
done